orchestra/tests/test_task_lifecycle.py

```diff
@@ -172,117 +177,246 @@ def test_worker_has_reviewer_status(self):
-        self.assertEquals(aborted_tasks.count(), 1)
-        aborted_task = aborted_tasks.first()
-        with self.assertRaises(TaskStatusError):
-            aborted_task = assign_task(self.workers[5].id, aborted_task.id)
-        self.assertEquals(aborted_task.assignments.count(), 1)
-        self.assertEqual(current_assignment(aborted_task).worker,
-                         self.workers[4])
+            reassign_assignment(self.workers[0].id, first_review_assignment.id)
+
+        # Reassign review assignment to another reviewer
+        first_review_assignment = task.assignments.get(assignment_counter=2)
+        entry_assignment = reassign_assignment(
+            self.workers[3].id, entry_assignment.id)
+        self.assertEquals(entry_assignment.worker, self.workers[3])
+
+    def test_revert_task_to_datetime(self):
+        test_start = timezone.now()
+
+        def _sample_complete_task():
+            task = Task.objects.create(
+                project=self.projects['empty_project'],
+                status=Task.Status.COMPLETE,
+                step=self.test_step,
+                start_datetime=test_start)
+
+            workers = {
+                'entry': self.workers[0],
+                'reviewer': self.workers[1]
+            }
+
+            assignments = [
+                {
+                    'worker': workers['entry'],
+                    'start_datetime': test_start + timedelta(hours=1)
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'start_datetime': test_start + timedelta(hours=3)
+                }
+            ]
+
+            snapshots = [
+                {
+                    'worker': workers['entry'],
+                    'type': TaskAssignment.SnapshotType.SUBMIT,
+                    'datetime': test_start + timedelta(hours=2),
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'type': TaskAssignment.SnapshotType.REJECT,
+                    'datetime': test_start + timedelta(hours=4),
+                },
+                {
+                    'worker': workers['entry'],
+                    'type': TaskAssignment.SnapshotType.SUBMIT,
+                    'datetime': test_start + timedelta(hours=5),
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'type': TaskAssignment.SnapshotType.ACCEPT,
+                    'datetime': test_start + timedelta(hours=6),
+                },
+            ]
+
+            for i, assignment in enumerate(assignments):
+                TaskAssignment.objects.create(
+                    worker=assignment['worker'], task=task,
+                    status=TaskAssignment.Status.SUBMITTED,
+                    in_progress_task_data={},
+                    assignment_counter=i,
+                    snapshots={'snapshots': [], '__version__': 2},
```

Comment:
use this function: https://github.com/unlimitedlabs/orchestra/blob/master/orchestra/utils/assignment_snapshots.py#L14